<!-- HTML header for doxygen 1.8.13-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<link rel="canonical" href="https://www.dealii.org/current/doxygen/deal.II/namespacedeal__II__exceptions_1_1internals.html" />
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.13"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>The deal.II Library: deal_II_exceptions::internals Namespace Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    extensions: ["tex2jax.js", "TeX/AMSmath.js", "TeX/AMSsymbols.js"],
    jax: ["input/TeX","output/HTML-CSS"],
});
</script><script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
<link href="stylesheet.css" rel="stylesheet" type="text/css"/>
<link rel="SHORTCUT ICON" href="deal.ico"></link>
<script type="text/javascript" src="custom.js"></script>
<meta name="author" content="The deal.II Authors <authors@dealii.org>"></meta>
<meta name="copyright" content="Copyright (C) 1998 - 2021 by the deal.II authors"></meta>
<meta name="deal.II-version" content="10.0.0-pre"></meta>
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="logo200.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">
   &#160;<span id="projectnumber">Reference documentation for deal.II version 10.0.0-pre</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!--Extra macros for MathJax:-->
<div style="display:none">
\(\newcommand{\dealvcentcolon}{\mathrel{\mathop{:}}}\)
\(\newcommand{\dealcoloneq}{\dealvcentcolon\mathrel{\mkern-1.2mu}=}\)
\(\newcommand{\jump}[1]{\left[\!\left[ #1 \right]\!\right]}\)
\(\newcommand{\average}[1]{\left\{\!\left\{ #1 \right\}\!\right\}}\)
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.13 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="namespacedeal__II__exceptions.html">deal_II_exceptions</a></li><li class="navelem"><a class="el" href="namespacedeal__II__exceptions_1_1internals.html">internals</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#enum-members">Enumerations</a> &#124;
<a href="#func-members">Functions</a> &#124;
<a href="#var-members">Variables</a>  </div>
  <div class="headertitle">
<div class="title">deal_II_exceptions::internals Namespace Reference<div class="ingroups"><a class="el" href="group__Exceptions.html">Exceptions and assertions</a></div></div>  </div>
</div><!--header-->
<div class="contents">
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="enum-members"></a>
Enumerations</h2></td></tr>
<tr class="memitem:a5dabff80d05c3f98e8e46f1068348d77"><td class="memItemLeft" align="right" valign="top">enum &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="namespacedeal__II__exceptions_1_1internals.html#a5dabff80d05c3f98e8e46f1068348d77">ExceptionHandling</a> { <a class="el" href="namespacedeal__II__exceptions_1_1internals.html#a5dabff80d05c3f98e8e46f1068348d77a8375e4e104f6f3742e93ac46cf89e6c3">abort_or_throw_on_exception</a>, 
<a class="el" href="namespacedeal__II__exceptions_1_1internals.html#a5dabff80d05c3f98e8e46f1068348d77a4cda1fb06c4e4bbf43a70adaa24ec1be">throw_on_exception</a>
 }</td></tr>
<tr class="separator:a5dabff80d05c3f98e8e46f1068348d77"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="func-members"></a>
Functions</h2></td></tr>
<tr class="memitem:a600f8f191a6ce368afda0074dd7ea1dc"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="namespacedeal__II__exceptions_1_1internals.html#a600f8f191a6ce368afda0074dd7ea1dc">abort</a> (const <a class="el" href="classExceptionBase.html">ExceptionBase</a> &amp;exc) noexcept</td></tr>
<tr class="separator:a600f8f191a6ce368afda0074dd7ea1dc"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a262c3f6ed8ff5ed894edbada5541a829"><td class="memTemplParams" colspan="2">template&lt;class ExceptionType &gt; </td></tr>
<tr class="memitem:a262c3f6ed8ff5ed894edbada5541a829"><td class="memTemplItemLeft" align="right" valign="top">void&#160;</td><td class="memTemplItemRight" valign="bottom"><a class="el" href="namespacedeal__II__exceptions_1_1internals.html#a262c3f6ed8ff5ed894edbada5541a829">issue_error_noreturn</a> (<a class="el" href="namespacedeal__II__exceptions_1_1internals.html#a5dabff80d05c3f98e8e46f1068348d77">ExceptionHandling</a> handling, const char *file, <a class="el" href="classint.html">int</a> line, const char *function, const char *cond, const char *exc_name, ExceptionType e)</td></tr>
<tr class="separator:a262c3f6ed8ff5ed894edbada5541a829"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab2c8e6dcb61cc485415d3e4dc6b601bf"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="namespacedeal__II__exceptions_1_1internals.html#ab2c8e6dcb61cc485415d3e4dc6b601bf">do_issue_error_nothrow</a> (const <a class="el" href="classExceptionBase.html">ExceptionBase</a> &amp;e) noexcept</td></tr>
<tr class="separator:ab2c8e6dcb61cc485415d3e4dc6b601bf"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af497c2ab12e991e1297f9f31f826ff07"><td class="memTemplParams" colspan="2">template&lt;class ExceptionType &gt; </td></tr>
<tr class="memitem:af497c2ab12e991e1297f9f31f826ff07"><td class="memTemplItemLeft" align="right" valign="top">void&#160;</td><td class="memTemplItemRight" valign="bottom"><a class="el" href="namespacedeal__II__exceptions_1_1internals.html#af497c2ab12e991e1297f9f31f826ff07">issue_error_nothrow</a> (const char *file, <a class="el" href="classint.html">int</a> line, const char *function, const char *cond, const char *exc_name, ExceptionType e) noexcept</td></tr>
<tr class="separator:af497c2ab12e991e1297f9f31f826ff07"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2248e7b7933509f19a2d21fc56ea3897"><td class="memItemLeft" align="right" valign="top">std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="namespacedeal__II__exceptions_1_1internals.html#a2248e7b7933509f19a2d21fc56ea3897">get_cusparse_error_string</a> (const cusparseStatus_t error_code)</td></tr>
<tr class="separator:a2248e7b7933509f19a2d21fc56ea3897"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a043abb5d484eed769834f982e5f16246"><td class="memItemLeft" align="right" valign="top">std::string&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="namespacedeal__II__exceptions_1_1internals.html#a043abb5d484eed769834f982e5f16246">get_cusolver_error_string</a> (const cusolverStatus_t error_code)</td></tr>
<tr class="separator:a043abb5d484eed769834f982e5f16246"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a195c4a77c41e0e0ba3392eb7e5541a24"><td class="memItemLeft" align="right" valign="top">std::string &amp;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="namespacedeal__II__exceptions_1_1internals.html#a195c4a77c41e0e0ba3392eb7e5541a24">get_additional_assert_output</a> ()</td></tr>
<tr class="separator:a195c4a77c41e0e0ba3392eb7e5541a24"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="var-members"></a>
Variables</h2></td></tr>
<tr class="memitem:acca0bc9b6db3845f927c12203f124717"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classbool.html">bool</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="namespacedeal__II__exceptions_1_1internals.html#acca0bc9b6db3845f927c12203f124717">allow_abort_on_exception</a> = true</td></tr>
<tr class="separator:acca0bc9b6db3845f927c12203f124717"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a17c4a70860f8fecf216cf8860e3bd98c"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classbool.html">bool</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="namespacedeal__II__exceptions_1_1internals.html#a17c4a70860f8fecf216cf8860e3bd98c">show_stacktrace</a> = true</td></tr>
<tr class="separator:a17c4a70860f8fecf216cf8860e3bd98c"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>这个命名空间中的函数与Assert和AssertThrow机制有关，但仅用于内部目的，不能在异常处理和抛出机制之外使用。 </p>
</div><h2 class="groupheader">Enumeration Type Documentation</h2>
<a id="a5dabff80d05c3f98e8e46f1068348d77"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a5dabff80d05c3f98e8e46f1068348d77">&#9670;&nbsp;</a></span>ExceptionHandling</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">enum <a class="el" href="namespacedeal__II__exceptions_1_1internals.html#a5dabff80d05c3f98e8e46f1068348d77">deal_II_exceptions::internals::ExceptionHandling</a></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>一个描述如何处理issue_error_noreturn中的异常的枚举。 </p>
<table class="fieldtable">
<tr><th colspan="2">Enumerator</th></tr><tr><td class="fieldname"><a id="a5dabff80d05c3f98e8e46f1068348d77a8375e4e104f6f3742e93ac46cf89e6c3"></a>abort_or_throw_on_exception&#160;</td><td class="fielddoc"><p>通过调用 <code>std::abort</code> 中止程序，除非 <a class="el" href="namespacedeal__II__exceptions.html#a3953411a9013d21af68e53fa5bbd4f8e">deal_II_exceptions::disable_abort_on_exception</a> 已经被调用：在这种情况下，程序将抛出一个异常。 </p>
</td></tr>
<tr><td class="fieldname"><a id="a5dabff80d05c3f98e8e46f1068348d77a4cda1fb06c4e4bbf43a70adaa24ec1be"></a>throw_on_exception&#160;</td><td class="fielddoc"><p>正常抛出异常。 </p>
</td></tr>
</table>

<p class="definition">Definition at line <a class="el" href="include_2deal_8II_2base_2exceptions_8h_source.html#l01268">1268</a> of file <a class="el" href="include_2deal_8II_2base_2exceptions_8h_source.html">exceptions.h</a>.</p>

</div>
</div>
<h2 class="groupheader">Function Documentation</h2>
<a id="a600f8f191a6ce368afda0074dd7ea1dc"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a600f8f191a6ce368afda0074dd7ea1dc">&#9670;&nbsp;</a></span>abort()</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void deal_II_exceptions::internals::abort </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classExceptionBase.html">ExceptionBase</a> &amp;&#160;</td>
          <td class="paramname"><em>exc</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">noexcept</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>通过打印 <code>exc</code> 提供的错误信息并调用 <code><a class="el" href="namespacedeal__II__exceptions_1_1internals.html#a600f8f191a6ce368afda0074dd7ea1dc">std::abort()</a></code>. 中止程序。 </p>

<p class="definition">Definition at line <a class="el" href="base_2exceptions_8cc_source.html#l00449">449</a> of file <a class="el" href="base_2exceptions_8cc_source.html">exceptions.cc</a>.</p>

</div>
</div>
<a id="a262c3f6ed8ff5ed894edbada5541a829"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a262c3f6ed8ff5ed894edbada5541a829">&#9670;&nbsp;</a></span>issue_error_noreturn()</h2>

<div class="memitem">
<div class="memproto">
<div class="memtemplate">
template&lt;class ExceptionType &gt; </div>
      <table class="memname">
        <tr>
          <td class="memname">void deal_II_exceptions::internals::issue_error_noreturn </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="namespacedeal__II__exceptions_1_1internals.html#a5dabff80d05c3f98e8e46f1068348d77">ExceptionHandling</a>&#160;</td>
          <td class="paramname"><em>handling</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>file</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classint.html">int</a>&#160;</td>
          <td class="paramname"><em>line</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>function</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>cond</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>exc_name</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">ExceptionType&#160;</td>
          <td class="paramname"><em>e</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>这个例程为<code>Assert</code>和<code>AssertThrow</code>宏中使用的异常生成机制做主要工作：正如其名称所暗示的，这个函数要么以抛出一个异常结束（如果 <code>handling</code> ]是throw_on_exception，或者 <code>handling</code> 是try_abort_exception且 <a class="el" href="namespacedeal__II__exceptions.html#a3953411a9013d21af68e53fa5bbd4f8e">deal_II_exceptions::disable_abort_on_exception</a> 是false），或者调用<code>abort</code>（如果 <code>handling</code> 是try_abort_exception且 <a class="el" href="namespacedeal__II__exceptions.html#a3953411a9013d21af68e53fa5bbd4f8e">deal_II_exceptions::disable_abort_on_exception</a> 是true）。 实际的异常对象（最后一个参数）通常是一个在原地创建的未命名的对象；因为我们修改了它，所以我们不能通过常量引用来获取它，而且临时变量不会与非常量引用绑定。 所以要用一个模板化的类型来取它的值（=copy it），以避免切分</p>
<ul>
<li>反正对性能的影响是相当小的。 <a class="el" href="classExceptionBase.html">ExceptionBase</a> </li>
</ul>

<p class="definition">Definition at line <a class="el" href="include_2deal_8II_2base_2exceptions_8h_source.html#l01306">1306</a> of file <a class="el" href="include_2deal_8II_2base_2exceptions_8h_source.html">exceptions.h</a>.</p>

</div>
</div>
<a id="ab2c8e6dcb61cc485415d3e4dc6b601bf"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ab2c8e6dcb61cc485415d3e4dc6b601bf">&#9670;&nbsp;</a></span>do_issue_error_nothrow()</h2>

<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void deal_II_exceptions::internals::do_issue_error_nothrow </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classExceptionBase.html">ExceptionBase</a> &amp;&#160;</td>
          <td class="paramname"><em>e</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">noexcept</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>内部函数，完成issue_error_nothrow的工作。 </p>

<p class="definition">Definition at line <a class="el" href="base_2exceptions_8cc_source.html#l00495">495</a> of file <a class="el" href="base_2exceptions_8cc_source.html">exceptions.cc</a>.</p>

</div>
</div>
<a id="af497c2ab12e991e1297f9f31f826ff07"></a>
<h2 class="memtitle"><span class="permalink"><a href="#af497c2ab12e991e1297f9f31f826ff07">&#9670;&nbsp;</a></span>issue_error_nothrow()</h2>

<div class="memitem">
<div class="memproto">
<div class="memtemplate">
template&lt;class ExceptionType &gt; </div>
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">void deal_II_exceptions::internals::issue_error_nothrow </td>
          <td>(</td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>file</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="classint.html">int</a>&#160;</td>
          <td class="paramname"><em>line</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>function</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>cond</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const char *&#160;</td>
          <td class="paramname"><em>exc_name</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">ExceptionType&#160;</td>
          <td class="paramname"><em>e</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">noexcept</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>异常生成机制，以防我们必须不抛出。 <a class="el" href="classExceptionBase.html">ExceptionBase</a> </p><dl class="section note"><dt>Note</dt><dd>这个函数是用一个模板定义的，原因与issue_error_noreturn()相同。 </dd></dl>

<p class="definition">Definition at line <a class="el" href="include_2deal_8II_2base_2exceptions_8h_source.html#l01354">1354</a> of file <a class="el" href="include_2deal_8II_2base_2exceptions_8h_source.html">exceptions.h</a>.</p>

</div>
</div>
<a id="a2248e7b7933509f19a2d21fc56ea3897"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a2248e7b7933509f19a2d21fc56ea3897">&#9670;&nbsp;</a></span>get_cusparse_error_string()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::string deal_II_exceptions::internals::get_cusparse_error_string </td>
          <td>(</td>
          <td class="paramtype">const cusparseStatus_t&#160;</td>
          <td class="paramname"><em>error_code</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>返回一个给定错误代码的字符串。这类似于cudaGetErrorString函数，但在cuSPARSE中没有相应的函数。 </p>

<p class="definition">Definition at line <a class="el" href="base_2exceptions_8cc_source.html#l00512">512</a> of file <a class="el" href="base_2exceptions_8cc_source.html">exceptions.cc</a>.</p>

</div>
</div>
<a id="a043abb5d484eed769834f982e5f16246"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a043abb5d484eed769834f982e5f16246">&#9670;&nbsp;</a></span>get_cusolver_error_string()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::string deal_II_exceptions::internals::get_cusolver_error_string </td>
          <td>(</td>
          <td class="paramtype">const cusolverStatus_t&#160;</td>
          <td class="paramname"><em>error_code</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>返回一个给定错误代码的字符串。这类似于cudaGetErrorString函数，但在cuSOLVER中没有相应的函数。 </p>

<p class="definition">Definition at line <a class="el" href="base_2exceptions_8cc_source.html#l00558">558</a> of file <a class="el" href="base_2exceptions_8cc_source.html">exceptions.cc</a>.</p>

</div>
</div>
<a id="a195c4a77c41e0e0ba3392eb7e5541a24"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a195c4a77c41e0e0ba3392eb7e5541a24">&#9670;&nbsp;</a></span>get_additional_assert_output()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::string&amp; deal_II_exceptions::internals::get_additional_assert_output </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p class="definition">Definition at line <a class="el" href="base_2exceptions_8cc_source.html#l00051">51</a> of file <a class="el" href="base_2exceptions_8cc_source.html">exceptions.cc</a>.</p>

</div>
</div>
<h2 class="groupheader">Variable Documentation</h2>
<a id="acca0bc9b6db3845f927c12203f124717"></a>
<h2 class="memtitle"><span class="permalink"><a href="#acca0bc9b6db3845f927c12203f124717">&#9670;&nbsp;</a></span>allow_abort_on_exception</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classbool.html">bool</a> deal_II_exceptions::internals::allow_abort_on_exception = true</td>
        </tr>
      </table>
</div><div class="memdoc">
<p>将这个变量设置为false将禁用deal.II的异常机制来中止问题。Assert()宏将抛出异常，而AssertNothrow()宏将只打印错误信息。这个变量不应该被直接改变。使用 <a class="el" href="namespacedeal__II__exceptions.html#a3953411a9013d21af68e53fa5bbd4f8e">disable_abort_on_exception()</a> 来代替。 </p>

<p class="definition">Definition at line <a class="el" href="base_2exceptions_8cc_source.html#l00059">59</a> of file <a class="el" href="base_2exceptions_8cc_source.html">exceptions.cc</a>.</p>

</div>
</div>
<a id="a17c4a70860f8fecf216cf8860e3bd98c"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a17c4a70860f8fecf216cf8860e3bd98c">&#9670;&nbsp;</a></span>show_stacktrace</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classbool.html">bool</a> deal_II_exceptions::internals::show_stacktrace = true</td>
        </tr>
      </table>
</div><div class="memdoc">

<p class="definition">Definition at line <a class="el" href="base_2exceptions_8cc_source.html#l00057">57</a> of file <a class="el" href="base_2exceptions_8cc_source.html">exceptions.cc</a>.</p>

</div>
</div>
</div><!-- contents -->
<!-- HTML footer for doxygen 1.8.13-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.13
</small></address>
</body>
</html>
